design/user_types.go

```diff
@@ -143,3 +143,317 @@ var identityData = a.Type("IdentityData", func() {
 	a.Attribute("attributes", identityDataAttributes, "Attributes of the user identity")
 	a.Required("type", "attributes")
 })
+
+//#############################################################################
+//
+// 			JSONAPI common
+//
+//#############################################################################
+
+// JSONAPILink represents a JSONAPI link object (see http://jsonapi.org/format/#document-links)
+var JSONAPILink = a.Type("JSONAPILink", func() {
+	a.Description(`See also http://jsonapi.org/format/#document-links.`)
+	a.Attribute("href", d.String, "a string containing the link's URL.", func() {
+		a.Example("http://example.com/articles/1/comments")
+	})
+	a.Attribute("meta", a.HashOf(d.String, d.Any), "a meta object containing non-standard meta-information about the link.")
+})
+
+// JSONAPIError represents a JSONAPI error object (see http://jsonapi.org/format/#error-objects)
+var JSONAPIError = a.Type("JSONAPIError", func() {
+	a.Description(`Error objects provide additional information about problems encountered while
+performing an operation. Error objects MUST be returned as an array keyed by errors in the
+top level of a JSON API document.
+
+See. also http://jsonapi.org/format/#error-objects.`)
+
+	a.Attribute("id", d.String, "a unique identifier for this particular occurrence of the problem.")
+	a.Attribute("links", a.HashOf(d.String, JSONAPILink), `a links object containing the following members:
+* about: a link that leads to further details about this particular occurrence of the problem.`)
+	a.Attribute("status", d.String, "the HTTP status code applicable to this problem, expressed as a string value.")
+	a.Attribute("code", d.String, "an application-specific error code, expressed as a string value.")
```

Comment:
The `status` is `404` but the `code` would be `not_found` for example. The structure is predefined by the JSONAPI spec. It wasn't my idea ;)